sequencer/src/context.rs

```diff
@@ -246,14 +255,136 @@ impl<N: ConnectedNetwork<PubKey>, P: SequencerPersistence, Ver: StaticVersionTyp
 
     /// Start participating in consensus.
     pub async fn start_consensus(&self) {
+        #[cfg(feature = "benchmarking")]
+        let mut has_orchestrator_client = false;
         if let Some(orchestrator_client) = &self.wait_for_orchestrator {
             tracing::warn!("waiting for orchestrated start");
             orchestrator_client
                 .wait_for_all_nodes_ready(self.node_state.node_id)
                 .await;
+            #[cfg(feature = "benchmarking")]
+            {
+                has_orchestrator_client = true;
+            }
+        } else {
+            tracing::error!("Cannot get info from orchestrator client");
         }
         tracing::warn!("starting consensus");
         self.handle.read().await.hotshot.start_consensus().await;
+
+        #[cfg(feature = "benchmarking")]
+        if has_orchestrator_client {
```

Comment:
+1 Yeah I'm also thinking about that

Reply:
I *think* you could just add a `benchmark()` method, gated by `benchmarking` feature to this same `impl`. Then just call that method on line 267 (instead of `has_orchestrator_client = true`). Then you wouldn't need the `has_orchestrator_client` variable or the following if statement that uses it.

Reply:
After the restructure (motivation here https://github.com/EspressoSystems/espresso-sequencer/issues/1695), I move all the benchmarking logic to `submit-transactions.rs`, since we already calculated latency there. So now they don't have their own function but they are together with the calculation after subscribing to `availability/stream/blocks/{}`.